src/lib/titled-hoc.jsx

```diff
@@ -12,26 +26,89 @@ const TitledHOC = function (WrappedComponent) {
             bindAll(this, [
                 'handleUpdateProjectTitle'
             ]);
-            this.state = {
-                projectTitle: null
-            };
+        }
+        componentDidMount () {
+            this.setReduxTitle(this.props.projectTitle);
+        }
+        componentDidUpdate (prevProps) {
+            if (this.props.projectTitle !== prevProps.projectTitle) {
+                this.setReduxTitle(this.props.projectTitle);
+            }
+            if (this.props.isShowingWithoutId && !prevProps.isShowingWithoutId) {
+                const defaultProjectTitle = this.titleWithDefault();
+                this.setReduxTitle(defaultProjectTitle);
+                this.props.onUpdateProjectTitle(defaultProjectTitle);
+            }
+        }
+        titleWithDefault (title) {
+            if (title === null || typeof title === 'undefined') {
+                return this.props.intl.formatMessage(messages.defaultProjectTitle);
+            }
+            return title;
+        }
+        setReduxTitle (newTitle) {
+            if (newTitle === null || typeof newTitle === 'undefined') {
+                this.props.onUpdateReduxProjectTitle(
+                    this.props.intl.formatMessage(messages.defaultProjectTitle)
+                );
+            } else {
+                this.props.onUpdateReduxProjectTitle(newTitle);
+            }
         }
         handleUpdateProjectTitle (newTitle) {
-            this.setState({projectTitle: newTitle});
+            this.setReduxTitle(newTitle);
+            this.props.onUpdateProjectTitle(newTitle);
         }
         render () {
+            const {
+                /* eslint-disable no-unused-vars */
+                intl,
+                isShowingWithoutId,
+                // for children, we replace onUpdateProjectTitle with our own
+                onUpdateProjectTitle,
+                onUpdateReduxProjectTitle,
+                // we don't pass projectTitle prop to children -- they must use
+                // redux value
+                projectTitle,
+                /* eslint-enable no-unused-vars */
+                ...componentProps
+            } = this.props;
             return (
                 <WrappedComponent
-                    canEditTitle
-                    projectTitle={this.state.projectTitle}
                     onUpdateProjectTitle={this.handleUpdateProjectTitle}
```

Comment:
If we expect children to receive the title from redux, I think we should we also expect them to set it through redux.

Reply:
I made this change, now testing it...

Reply:
Tested well